tests/cpptests/test_cpp_query_validation.cpp

```diff
@@ -17,30 +17,58 @@
 
 class QueryValidationTest : public ::testing::Test {};
 
-bool isValidAsVectorFilter(const char *qt, RedisSearchCtx &ctx) {
+bool isValidAsHybridVectorFilter(const char *qt, RedisSearchCtx &ctx) {
   QASTCXX ast;
   ast.setContext(&ctx);
-  return ast.isValidAsVectorFilter(qt);
+  return ast.isValidQuery(qt, QAST_HYBRID_VSIM_FILTER_CLAUSE);
 }
 
-#define assertValidVectorFilter(qt, ctx) ASSERT_TRUE(isValidAsVectorFilter(qt, ctx))
-#define assertInvalidVectorFilter(qt, ctx) ASSERT_FALSE(isValidAsVectorFilter(qt, ctx))
+#define assertValidHybridVectorFilter(qt, ctx) ASSERT_TRUE(isValidAsHybridVectorFilter(qt, ctx))
 
 bool isValidAsHybridSearch(const char *qt, RedisSearchCtx &ctx) {
   QASTCXX ast;
   ast.setContext(&ctx);
-  return ast.isValidAsHybridSearch(qt);
+  return ast.isValidQuery(qt, QAST_HYBRID_SEARCH_CLAUSE);
 }
+
+bool isInvalidHybridSearch(const char *qt, RedisSearchCtx &ctx,
+  QAST_ValidationFlags validationFlags, QueryErrorCode error) {
+  QASTCXX ast;
+  ast.setContext(&ctx);
+  // First check if the query is invalid
+  bool valid = ast.isValidQuery(qt, validationFlags);
+  // Then check if the error message contains the expected error
+  QueryErrorCode actual_err_code = ast.getErrorCode();
+  // If the query is valid or the error code doesn't match, the test should fail
+  if (valid) {
+    ADD_FAILURE() << "Query should be invalid but was valid: " << qt;
+    return false;
+  }
+  if (actual_err_code != error) {
+    ADD_FAILURE() << "Error code mismatch for query '" << qt
+                  << "': expected " << error
+                  << " but got " << actual_err_code
+                  << " Error message: " << ast.getError();
+    return false;
+  }
```

Comment:
Redundant?

Reply:
OK. I removed the `if(valid)` condition, for valid queries, the comparison of the error codes is enough to return `false`.